stdlib/3.7/dataclasses.pyi

```diff
@@ -0,0 +1,67 @@
+from typing import overload, Any, Callable, Dict, Generic, Iterable, Mapping, Optional, Tuple, Type, TypeVar, Union
+
+
+_T = TypeVar('_T')
+
+_DictFactory = Callable[List[Tuple[str, Any]], _T]
+_TupleFactory = Callable[List[Any], _T]
+
+class _MISSING_TYPE: ...
+MISSING: _MISSING_TYPE
+
+class _InitVarMeta(type): ...
+
+def asdict(obj: Any, *, dict_factory: _DictFactory = ...) -> _T: ...
+
+def astuple(obj: Any, *, tuple_factory: _TupleFactory = ...) -> _T: ...
+
+
+@overload
+def dataclass(_cls: Type[_T]) -> Type[_T]: ...
+
+@overload
+def dataclass(*, init: bool = ..., repr: bool = ..., eq: bool = ..., order: bool = ...,
+    unsafe_hash: bool = ..., frozen: bool = ...) -> Callable[[Type[_T]], Type[_T]]: ...
+
+
+class Field(Generic[_T]):
+    name: str
+    type: Type[_T]
+    default: _T
+    default_factory: Callable[[], _T]
+    repr: bool
+    hash: Optional[bool]
+    init: bool
+    compare: bool
+    metadata: Optional[Mapping[str, Any]]
+
+
+@overload # `default` and `default_factory` are optional and mutually exclusive.
+def field(*, default: _T = ..., default_factory: _MISSING_TYPE = ...,
```

Comment:
All three overloads are actually the same unless I'm missing something.

Reply:
Can you elaborate? The return type of the last one is `Field[Any]`, while the first two are constrained.